test/unittests/execute_floating_point_test.cpp

```diff
@@ -465,6 +465,62 @@ TEST(execute_floating_point, f64_promote_f32)
     EXPECT_GT(FP{res4.value.f64}.nan_payload(), FP64::canon);
 }
 
+TEST(execute_floating_point, f32_demote_f64)
+{
+    /* wat2wasm
+    (func (param f64) (result f32)
+      local.get 0
+      f32.demote_f64
+    )
+    */
+    const auto wasm = from_hex("0061736d0100000001060160017c017d030201000a070105002000b60b");
+    auto instance = instantiate(parse(wasm));
+
+    const std::pair<double, float> test_cases[] = {
+        // demote(+-0) = +-0
+        {0.0, 0.0f},
+        {-0.0, -0.0f},
+
+        {1.0, 1.0f},
+        {-1.0, -1.0f},
+        {double{FP32::Limits::lowest()}, FP32::Limits::lowest()},
+        {double{FP32::Limits::max()}, FP32::Limits::max()},
+        {double{FP32::Limits::min()}, FP32::Limits::min()},
+        {double{FP32::Limits::denorm_min()}, FP32::Limits::denorm_min()},
+
+        // Out of range values rounded to max/lowest.
+        {std::nextafter(double{FP32::Limits::max()}, FP64::Limits::infinity()),
+            FP32::Limits::max()},
+        {std::nextafter(double{FP32::Limits::lowest()}, -FP64::Limits::infinity()),
+            FP32::Limits::lowest()},
+        {double{FP32::Limits::max()} * 0x1.0000008000007p0, FP32::Limits::max()},
+        {double{FP32::Limits::lowest()} * 0x1.0000008000007p0, FP32::Limits::lowest()},
+
+        // The smallest of range values rounded to infinity.
+        {double{FP32::Limits::max()} * 0x1.0000008000008p0, FP32::Limits::infinity()},
+        {double{FP32::Limits::lowest()} * 0x1.0000008000008p0, -FP32::Limits::infinity()},
```

Comment:
What are these values?

Reply:
Boundaries values that convert to max or infinity. I will rework it to be more readable.